src/types/local_date.rs

```diff
@@ -0,0 +1,271 @@
+use crate::utils::js_error;
+use scylla::frame::value::CqlDate;
+use std::{cmp::max, fmt, num::ParseIntError};
+
+// Max and min date range of the Date class in JS.
+const MAX_JS_DATE: i32 = 100_000_000;
+const MIN_JS_DATE: i32 = -MAX_JS_DATE;
+
+// Number of leap years up to 1970.
+const LEAP_YEAR_1970: i32 = 477;
+// Number of days to the beginning of each month.
+const DAY_IN_MONTH: [i32; 12] = [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334];
+#[napi]
+pub struct LocalDateWrapper {
+    pub value: i32, // wrapper for number of days from 01.01.1970
+    pub year: Option<i32>,
+    pub month: Option<i8>,
+    pub day: Option<i8>,
+    pub in_date: bool, // value can be represented as Date class in JS
+}
+
+#[napi]
+impl LocalDateWrapper {
+    /// Create a new object from the day, month and year.
+    #[napi]
+    pub fn new(day: i8, month: i8, year: i32) -> napi::Result<Self> {
+        Self::check_valid_date(day, month, year)
+    }
+
+    /// Create a new object from number of days since 01.01.1970.
+    #[napi]
+    pub fn new_day(value: i32) -> napi::Result<Self> {
+        if !(MIN_JS_DATE..=MAX_JS_DATE).contains(&value) {
+            Ok(LocalDateWrapper {
+                value,
+                year: None,
+                month: None,
+                day: None,
+                in_date: false,
+            })
+        } else {
+            let date = Self::date_from_days(value.into());
+            Ok(LocalDateWrapper::new(date.0, date.1, date.2)?)
+        }
+    }
+
+    /// Counts the number of days since 01.01.1970.
+    fn days_since_epoch(day: i8, month: i8, year: i32) -> i32 {
+        let mut total_days = 0;
+
+        let number_day = DAY_IN_MONTH[month as usize - 1] // number of days from 1 January
+            + if Self::is_leap_year(year) && month > 2 {
+                1
+            } else {
+                0
+            }
+            + day as i32
+            - 1;
```

Comment:
❓ We offset the value two times by a constant. Maybe there is a cleaner way to write it?